numpy/core/tests/test_einsum_object.py

```diff
@@ -0,0 +1,309 @@
+import itertools
```

Comment:
Most of the test case are just copied from `test_einsum.py`, which I think is unnecessary, you can just add object type to `test_einsum.py`.

Reply:
Ah, that's another issue I got stuck on.

The tests in `check_einsum_sums` do somethig like:
```
a = np.arange(1, dtype='object')
np.sum(a, axis=-1).astype('object'))
```

which raises _`AttributeError: int' object has no attribute 'astype'
I guess a object array with size =1 gets converted to a regular python int at some point in np.sum()
So, I needed a test to remove the _.astype_

Reply:
You could always add a special case within the `test_einsum.py` tests for when `dtype=object`, to avoid those troublesome tests.

Reply:
> You could always add a special case within the test_einsum.py tests for when dtype=object, to avoid those troublesome tests.

Done